src/library/searchqueryparser.h

```diff
@@ -11,14 +11,19 @@
 
 class SearchQueryParser {
   public:
-    SearchQueryParser(QSqlDatabase& database);
+    SearchQueryParser(TrackCollection* pTrackCollection);
```

Comment:
If you could explain that a little more! I understand what explicit and implicit conversion is, but I don't really get your point here. Also by your logic the 
`SearchQueryParser(QSqlDatabase& database);` constructor that was there was also wrong and should be explicit right?